app/views/schools/_discord_roles_diff.html.erb

```diff
@@ -0,0 +1,52 @@
+<div class="mx-auto mt-6 max-w-6xl px-5">
+  <div class="rounded-xl border border-gray-200 bg-gray-50">
+    <div class="sticky top-0 bg-gray-50 flex flex-wrap items-center justify-between gap-5 border-b border-gray-200 p-4">
+      <div>
+        <h2 class="text-lg font-semibold"><%= t(".review_changes.title") %></h2>
+        <p class="text-sm text-gray-400 font-medium"><%= t(".review_changes.warn") %></p>
+      </div>
+      <div class="flex gap-4">
+        <%= link_to t(".cancel_sync"), discord_server_roles_school_path, class: "rounded-lg border border-gray-200 bg-gray-100 px-5 py-3 text-sm font-medium text-gray-500 transition hover:bg-white" %>
+        <%= button_to t(".confirm_sync"), discord_sync_roles_school_path(confirmed_sync: true), method: :post, class: "border-primary-200 bg-primary-100 text-primary-500 hover:bg-primary-200 rounded-lg border px-5 py-3 text-sm font-medium transition" %>
+      </div>
+    </div>
+    <div class="grid grid-cols-2">
+      <div class="border-e border-gray-200 p-4">
+        <h3 class="text-sm font-semibold"><%= t(".saved_roles_on_lms") %></h3>
+        <div class="mt-3 flex flex-col gap-2">
+          <% locals[:sync_service].cached_roles.each do |role| %>
```

Comment:
Passed in `locals` can be accessed as local variables. You don't need to take this hash-read approach unless the partial is expected to work both with and without these local variables.

```suggestion
          <% sync_service.cached_roles.each do |role| %>
```